rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/metadata/TbGetOriginatorFieldsNode.java

```diff
@@ -70,7 +70,7 @@ private ListenableFuture<Void> putEntityFields(TbContext ctx, EntityId entityId,
                     data -> {
                         config.getFieldsMapping().forEach((field, metaKey) -> {
                             String val = data.getFieldValue(field);
-                            if (val != null) {
+                            if (val != null && !val.equals("null")) {
```

Comment:
This fix is not backward compatible. We need to have a enable/disable checkbox for this logic. That should be disabled by default.